File: Python3/2303_calculate_amount_paid_in_taxes.py
# https://leetcode.cn/problems/calculate-amount-paid-in-taxes/

class Solution:
    def calculateTax(self, brackets, income):
        """
        Inputs:
            brackets: list[list[int]]
            income: int
        Outputs:
            res: float
        """
        res, lower = 0, 0
        for upper, rate in brackets:
            res += (min(income, upper) - lower) * rate
            if upper >= income:
                break
            lower = upper
        return res / 100

File: Cpp/2303_calculate_amount_paid_in_taxes.cpp
// https://leetcode.cn/problems/calculate-amount-paid-in-taxes/

class Solution {
public:
    double calculateTax(vector<vector<int>>& brackets, int income) {
        int res = 0, lower = 0;
        for (auto br: brackets) {
            int upper = br[0];
            res += (min(income, upper) - lower) * br[1];
            if (upper >= income) {
                break;
            }
            lower = upper;
        }
        return (double) res / 100.;
    }
};